Index: src/main/api/models/base_model.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/base_model.py b/src/main/api/models/base_model.py
new file mode 100644
--- /dev/null	(date 1765443991122)
+++ b/src/main/api/models/base_model.py	(date 1765443991122)
@@ -0,0 +1,4 @@
+from pydantic import BaseModel as BM
+
+class BaseModel(BM):
+    pass
Index: src/main/api/models/create_user_request.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/create_user_request.py b/src/main/api/models/create_user_request.py
new file mode 100644
--- /dev/null	(date 1766042308577)
+++ b/src/main/api/models/create_user_request.py	(date 1766042308577)
@@ -0,0 +1,9 @@
+from typing import Annotated
+
+from src.main.api.generators.creation_rule import CreationRule
+from src.main.api.models.base_model import BaseModel
+
+class CreateUserRequest(BaseModel):
+    username: Annotated[str, CreationRule(regex=r'^[A-Za-z0-9]{3,15}$')]
+    password: Annotated[str, CreationRule(regex=r'^[A-Z]{3}[a-z]{1}[0-9]{2}[!?@#%$_]{4}$')]
+    role: Annotated[str, CreationRule(regex=r'^ROLE_USER')]
\ No newline at end of file
Index: src/main/api/models/create_user_reesponse.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/create_user_reesponse.py b/src/main/api/models/create_user_reesponse.py
new file mode 100644
--- /dev/null	(date 1765445587921)
+++ b/src/main/api/models/create_user_reesponse.py	(date 1765445587921)
@@ -0,0 +1,8 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class CreateUserResponse(BaseModel):
+    id: int
+    username: str
+    password: str
+    role: str
\ No newline at end of file
Index: src/main/api/models/login_user_response.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/login_user_response.py b/src/main/api/models/login_user_response.py
new file mode 100644
--- /dev/null	(date 1765446432283)
+++ b/src/main/api/models/login_user_response.py	(date 1765446432283)
@@ -0,0 +1,10 @@
+from src.main.api.models.base_model import BaseModel
+
+class User(BaseModel):
+    username:str
+    role:str
+
+
+class LoginUserResponse(BaseModel):
+    token:  str
+    user: User
\ No newline at end of file
Index: src/main/api/models/create_account_response.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/create_account_response.py b/src/main/api/models/create_account_response.py
new file mode 100644
--- /dev/null	(date 1765444673188)
+++ b/src/main/api/models/create_account_response.py	(date 1765444673188)
@@ -0,0 +1,7 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class CreateAccountResponse(BaseModel):
+    id:int
+    number:str
+    balance:float
Index: src/main/api/requests/requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/requester.py b/src/main/api/requests/requester.py
new file mode 100644
--- /dev/null	(date 1765532345251)
+++ b/src/main/api/requests/requester.py	(date 1765532345251)
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+from typing import Dict, Callable
+from src.main.api.models.base_model import BaseModel
+
+
+class Requester(ABC):
+    def __init__(self, request_spec: Dict[str, str], response_spec: Callable):
+        self.headers = request_spec['headers']
+        self.base_url = request_spec['base_url']
+        self.response_spec = response_spec
+
+    @abstractmethod
+    def post(self, model:BaseModel):...
Index: src/main/api/requests/create_user_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/create_user_requester.py b/src/main/api/requests/create_user_requester.py
new file mode 100644
--- /dev/null	(date 1765717632787)
+++ b/src/main/api/requests/create_user_requester.py	(date 1765717632787)
@@ -0,0 +1,20 @@
+from http import HTTPStatus
+from requests import Response
+from src.main.api.models.create_user_reesponse import CreateUserResponse
+from src.main.api.models.create_user_request import CreateUserRequest
+from src.main.api.requests.requester import Requester
+import requests
+
+
+class CreateUserRequester(Requester):
+    def post(self, create_user_request: CreateUserRequest) -> CreateUserResponse|Response:
+        url = f'{self.base_url}/admin/create'
+        response = requests.post(
+            url = url,
+            json = create_user_request.model_dump(),
+            headers=self.headers
+        )
+        self.response_spec(response)
+        if response.status_code in [200, 201]:
+            return CreateUserResponse(**response.json())
+        return response
Index: src/main/api/requests/login_user_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/login_user_requester.py b/src/main/api/requests/login_user_requester.py
new file mode 100644
--- /dev/null	(date 1765717561274)
+++ b/src/main/api/requests/login_user_requester.py	(date 1765717561274)
@@ -0,0 +1,18 @@
+import requests
+from requests import Response
+from src.main.api.models.login_user_response import LoginUserResponse
+from src.main.api.requests.requester import Requester
+from src.main.api.models.login_user_request import LoginUserRequest
+
+class LoginUserRequester(Requester):
+    def post(self, login_user_request:LoginUserRequest) -> LoginUserResponse|Response:
+        url = f'{self.base_url}/auth/token/login'
+        response = requests.post(
+            url=url,
+            json=login_user_request.model_dump(),
+            headers=self.headers
+        )
+        self.response_spec(response)
+        if response.status_code == 200:
+            return LoginUserResponse(**response.json())
+        return response
\ No newline at end of file
Index: src/main/api/requests/create_account_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/create_account_requester.py b/src/main/api/requests/create_account_requester.py
new file mode 100644
--- /dev/null	(date 1765708289625)
+++ b/src/main/api/requests/create_account_requester.py	(date 1765708289625)
@@ -0,0 +1,13 @@
+from src.main.api.models.create_account_response import CreateAccountResponse
+from src.main.api.requests.requester import Requester
+import requests
+
+class CreateAccountRequest(Requester):
+    def post(self, model = None) -> CreateAccountResponse:
+        url = f'{self.base_url}/account/create'
+        response = requests.post(
+            url = url,
+            headers=self.headers,
+        )
+        self.response_spec(response)
+        return CreateAccountResponse(**response.json())
Index: resources/urls.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/resources/urls.properties b/resources/urls.properties
new file mode 100644
--- /dev/null	(date 1765543239513)
+++ b/resources/urls.properties	(date 1765543239513)
@@ -0,0 +1,1 @@
+backendUrl=http://localhost:4111/api
\ No newline at end of file
Index: src/main/api/configs/config.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/configs/config.py b/src/main/api/configs/config.py
new file mode 100644
--- /dev/null	(date 1765544374572)
+++ b/src/main/api/configs/config.py	(date 1765544374572)
@@ -0,0 +1,27 @@
+from pathlib import Path
+from typing import Any
+
+class Config:
+    _isinstance = None
+    _dictionary = {}
+
+    def __new__(cls):
+        if cls._isinstance is None:
+            cls._isinstance = super(Config, cls).__new__(cls)
+
+            config_path = Path(__file__).parents[4] / 'resources' / 'urls.properties'
+
+            if not config_path.exists():
+                raise FileNotFoundError(f'Config path not found: {config_path}')
+
+            with open(config_path, 'r') as f:
+                for line in f:
+                    if '=' in line:
+                        key,value = line.split('=')
+                        cls._dictionary[key] = value
+
+        return cls._isinstance
+
+    @staticmethod
+    def fetch(key:str, default_value: Any = None) -> Any:
+        return Config()._dictionary.get(key, default_value)
\ No newline at end of file
Index: .idea/inspectionProfiles/Project_Default.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/inspectionProfiles/Project_Default.xml b/.idea/inspectionProfiles/Project_Default.xml
new file mode 100644
--- /dev/null	(date 1765434922916)
+++ b/.idea/inspectionProfiles/Project_Default.xml	(date 1765434922916)
@@ -0,0 +1,82 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="PyPackageRequirementsInspection" enabled="false" level="WARNING" enabled_by_default="false">
+      <option name="ignoredPackages">
+        <list>
+          <option value="alembic" />
+          <option value="annotated-doc" />
+          <option value="annotated-types" />
+          <option value="anyio" />
+          <option value="asyncpg" />
+          <option value="certifi" />
+          <option value="charset-normalizer" />
+          <option value="click" />
+          <option value="colorama" />
+          <option value="dnspython" />
+          <option value="email-validator" />
+          <option value="fastapi" />
+          <option value="fastapi-cli" />
+          <option value="fastapi-cloud-cli" />
+          <option value="greenlet" />
+          <option value="h11" />
+          <option value="httpcore" />
+          <option value="httptools" />
+          <option value="httpx" />
+          <option value="idna" />
+          <option value="iniconfig" />
+          <option value="Jinja2" />
+          <option value="Mako" />
+          <option value="markdown-it-py" />
+          <option value="MarkupSafe" />
+          <option value="mdurl" />
+          <option value="packaging" />
+          <option value="pluggy" />
+          <option value="psycopg" />
+          <option value="psycopg-binary" />
+          <option value="pydantic" />
+          <option value="pydantic-settings" />
+          <option value="pydantic_core" />
+          <option value="Pygments" />
+          <option value="pytest" />
+          <option value="python-dotenv" />
+          <option value="python-multipart" />
+          <option value="PyYAML" />
+          <option value="requests" />
+          <option value="rich" />
+          <option value="rich-toolkit" />
+          <option value="rignore" />
+          <option value="sentry-sdk" />
+          <option value="shellingham" />
+          <option value="sniffio" />
+          <option value="SQLAlchemy" />
+          <option value="starlette" />
+          <option value="typer" />
+          <option value="typing-inspection" />
+          <option value="typing_extensions" />
+          <option value="tzdata" />
+          <option value="urllib3" />
+          <option value="utils" />
+          <option value="uvicorn" />
+          <option value="uvloop" />
+          <option value="watchfiles" />
+          <option value="websockets" />
+        </list>
+      </option>
+    </inspection_tool>
+    <inspection_tool class="PyPep8NamingInspection" enabled="true" level="WEAK WARNING" enabled_by_default="true">
+      <option name="ignoredErrors">
+        <list>
+          <option value="N802" />
+        </list>
+      </option>
+    </inspection_tool>
+    <inspection_tool class="PyUnresolvedReferencesInspection" enabled="true" level="WARNING" enabled_by_default="true">
+      <option name="ignoredIdentifiers">
+        <list>
+          <option value="_collections_abc.dict_keys.*" />
+        </list>
+      </option>
+    </inspection_tool>
+  </profile>
+</component>
\ No newline at end of file
Index: src/main/api/tests/create_deposit_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/create_deposit_test.py b/src/main/api/tests/create_deposit_test.py
new file mode 100644
--- /dev/null	(date 1766062127367)
+++ b/src/main/api/tests/create_deposit_test.py	(date 1766062127367)
@@ -0,0 +1,38 @@
+from src.main.api.models.deposit_request import DepositRequest
+import pytest
+
+@pytest.mark.api
+class TestCreateDepositTest:
+    # username = 'Stepans031122'
+    # def test_replenishment_deposit(self):
+    #     create_user_request = CreateUserRequest(username=TestCreateDepositTest.username, password='Pas!sw0rd', role='ROLE_USER')
+    #     CreateUserRequester(
+    #         request_spec=RequestSpecs.auth_headers(username='admin', password='123456'),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(create_user_request)
+    #
+    #     create_account_response = CreateAccountRequest(
+    #         request_spec=RequestSpecs.auth_headers(username=create_user_request.username, password='Pas!sw0rd'),
+    #         response_spec=ResponseSpecs.request_create()
+    #     ).post()
+    #
+    #     account_id = create_account_response.id
+    #     assert create_account_response.balance == 0
+    #
+    #
+    #     deposit_request = DepositRequest(accountId=account_id, amount=3000)
+    #     create_deposit_response = CreateDepositRequester(
+    #         request_spec=RequestSpecs.auth_headers(create_user_request.username, create_user_request.password),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(deposit_request)
+    #     assert create_deposit_response.balance == 3000
+    #     print(create_deposit_response)
+    def test_deposit(self, api_manager, create_user_request):
+        account_id = api_manager.user_steps.create_account(create_user_request).id
+        deposit_request = DepositRequest(accountId=account_id, amount=5500)
+        create_deposit = api_manager.user_steps.create_deposit(create_user_request, deposit_request)
+
+        assert create_deposit.balance == 5500
+
+
+
Index: .idea/NewSwaggerTest.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/NewSwaggerTest.iml b/.idea/NewSwaggerTest.iml
new file mode 100644
--- /dev/null	(date 1765434924347)
+++ b/.idea/NewSwaggerTest.iml	(date 1765434924347)
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/.venv" />
+    </content>
+    <orderEntry type="jdk" jdkName="Python 3.13 (NewSwaggerTest)" jdkType="Python SDK" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1765435068677)
+++ b/.idea/misc.xml	(date 1765435068677)
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="Black">
+    <option name="sdkName" value="Python 3.13 (NewSwaggerTest)" />
+  </component>
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.13 (NewSwaggerTest)" project-jdk-type="Python SDK" />
+</project>
\ No newline at end of file
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/modules.xml b/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1765434923111)
+++ b/.idea/modules.xml	(date 1765434923111)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/NewSwaggerTest.iml" filepath="$PROJECT_DIR$/.idea/NewSwaggerTest.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: .idea/workspace.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/workspace.xml b/.idea/workspace.xml
new file mode 100644
--- /dev/null	(date 1766062775916)
+++ b/.idea/workspace.xml	(date 1766062775916)
@@ -0,0 +1,224 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="AutoImportSettings">
+    <option name="autoReloadType" value="SELECTIVE" />
+  </component>
+  <component name="ChangeListManager">
+    <list default="true" id="a63f5bb8-0c26-4d2b-af29-c8932684b6db" name="Changes" comment="">
+      <change afterPath="$PROJECT_DIR$/.idea/NewSwaggerTest.iml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/.idea/inspectionProfiles/Project_Default.xml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/.idea/inspectionProfiles/profiles_settings.xml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/.idea/misc.xml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/.idea/modules.xml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/.idea/vcs.xml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/.idea/workspace.xml" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/conftest.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/pytest.ini" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/resources/urls.properties" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/classes/api_manager.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/configs/config.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/fixtures/api_fixture.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/fixtures/object_fixture.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/fixtures/user_fixture.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/foundation/crud_endpoint.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/foundation/endpoint.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/foundation/http_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/foundation/requesters/crud_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/foundation/requesters/validate_crud_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/generators/creation_rule.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/generators/model_generator.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/base_model.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/create_account_response.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/create_user_reesponse.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/create_user_request.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/credit_request.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/credit_response.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/deposit_request.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/deposit_response.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/login_user_request.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/login_user_response.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/repay_credit_request.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/repay_credit_response.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/transfer_request.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/models/transfer_response.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/create_account_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/create_user_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/credit_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/deposit_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/login_user_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/repay_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/requests/transfer_requester.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/specs/request_specs.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/specs/response_specs.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/steps/admin_steps.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/steps/base_steps.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/steps/user_steps.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/create_account_test.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/create_credit_request_test.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/create_deposit_test.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/create_transfer_test.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/create_user_test.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/repay_credit_test.py" afterDir="false" />
+      <change afterPath="$PROJECT_DIR$/src/main/api/tests/user_login_test.py" afterDir="false" />
+    </list>
+    <option name="SHOW_DIALOG" value="false" />
+    <option name="HIGHLIGHT_CONFLICTS" value="true" />
+    <option name="HIGHLIGHT_NON_ACTIVE_CHANGELIST" value="false" />
+    <option name="LAST_RESOLUTION" value="IGNORE" />
+  </component>
+  <component name="FileTemplateManagerImpl">
+    <option name="RECENT_TEMPLATES">
+      <list>
+        <option value="Python Script" />
+      </list>
+    </option>
+  </component>
+  <component name="Git.Settings">
+    <option name="RECENT_GIT_ROOT_PATH" value="$PROJECT_DIR$" />
+  </component>
+  <component name="ProjectColorInfo">{
+  &quot;associatedIndex&quot;: 0
+}</component>
+  <component name="ProjectId" id="36gj50H25KuBYf58ySFg8FCSeaS" />
+  <component name="ProjectViewState">
+    <option name="hideEmptyMiddlePackages" value="true" />
+    <option name="showLibraryContents" value="true" />
+  </component>
+  <component name="PropertiesComponent"><![CDATA[{
+  "keyToString": {
+    "ASKED_SHARE_PROJECT_CONFIGURATION_FILES": "true",
+    "ModuleVcsDetector.initialDetectionPerformed": "true",
+    "Python tests.Python tests for src.main.api.tests.create_account_test.TestCreateAccount.test_create_account.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.test_request_credit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.test_request_credit_level_2.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.test_request_credit_senior.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_deposit_test.CreateDepositTest.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_deposit_test.CreateDepositTest.test_replenishment_deposit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_deposit_test.TestCreateDepositTest.test_deposit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_deposit_test.TestCreateDepositTest.test_replenishment_deposit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_transfer_test.TestCreateDepositTest.test_replenishment_deposit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_transfer_test.TestCreateTransfer.test_transfer_deposit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_transfer_test.TestCreateTransfer.test_transfer_deposit_level_2.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_user_test.TestCreateUser.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_user_test.TestCreateUser.test_create_user_invalid.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.create_user_test.TestCreateUser.test_create_user_valid.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit_level_2.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit_senior.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.user_login_test.TestUserLogin.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.user_login_test.TestUserLogin.test_login_admin.executor": "Run",
+    "Python tests.Python tests for src.main.api.tests.user_login_test.TestUserLogin.test_user_login.executor": "Run",
+    "RunOnceActivity.ShowReadmeOnStart": "true",
+    "RunOnceActivity.TerminalTabsStorage.copyFrom.TerminalArrangementManager.252": "true",
+    "RunOnceActivity.git.unshallow": "true",
+    "SHARE_PROJECT_CONFIGURATION_FILES": "true",
+    "git-widget-placeholder": "master",
+    "settings.editor.selected.configurable": "org.jetbrains.plugins.github.ui.GithubSettingsConfigurable"
+  }
+}]]></component>
+  <component name="RunManager" selected="Python tests.Python tests for src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit_senior">
+    <configuration name="Python tests for src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.test_request_credit_senior" type="tests" factoryName="Autodetect" temporary="true" nameIsGenerated="true">
+      <module name="NewSwaggerTest" />
+      <option name="ENV_FILES" value="" />
+      <option name="INTERPRETER_OPTIONS" value="" />
+      <option name="PARENT_ENVS" value="true" />
+      <option name="SDK_HOME" value="" />
+      <option name="WORKING_DIRECTORY" value="$PROJECT_DIR$" />
+      <option name="IS_MODULE_SDK" value="true" />
+      <option name="ADD_CONTENT_ROOTS" value="true" />
+      <option name="ADD_SOURCE_ROOTS" value="true" />
+      <option name="_new_additionalArguments" value="&quot;&quot;" />
+      <option name="_new_target" value="&quot;src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.test_request_credit_senior&quot;" />
+      <option name="_new_targetType" value="&quot;PYTHON&quot;" />
+      <method v="2" />
+    </configuration>
+    <configuration name="Python tests for src.main.api.tests.create_deposit_test.TestCreateDepositTest.test_deposit" type="tests" factoryName="Autodetect" temporary="true" nameIsGenerated="true">
+      <module name="NewSwaggerTest" />
+      <option name="ENV_FILES" value="" />
+      <option name="INTERPRETER_OPTIONS" value="" />
+      <option name="PARENT_ENVS" value="true" />
+      <option name="SDK_HOME" value="" />
+      <option name="WORKING_DIRECTORY" value="$PROJECT_DIR$" />
+      <option name="IS_MODULE_SDK" value="true" />
+      <option name="ADD_CONTENT_ROOTS" value="true" />
+      <option name="ADD_SOURCE_ROOTS" value="true" />
+      <option name="_new_additionalArguments" value="&quot;&quot;" />
+      <option name="_new_target" value="&quot;src.main.api.tests.create_deposit_test.TestCreateDepositTest.test_deposit&quot;" />
+      <option name="_new_targetType" value="&quot;PYTHON&quot;" />
+      <method v="2" />
+    </configuration>
+    <configuration name="Python tests for src.main.api.tests.create_transfer_test.TestCreateTransfer.test_transfer_deposit" type="tests" factoryName="Autodetect" temporary="true" nameIsGenerated="true">
+      <module name="NewSwaggerTest" />
+      <option name="ENV_FILES" value="" />
+      <option name="INTERPRETER_OPTIONS" value="" />
+      <option name="PARENT_ENVS" value="true" />
+      <option name="SDK_HOME" value="" />
+      <option name="WORKING_DIRECTORY" value="$PROJECT_DIR$" />
+      <option name="IS_MODULE_SDK" value="true" />
+      <option name="ADD_CONTENT_ROOTS" value="true" />
+      <option name="ADD_SOURCE_ROOTS" value="true" />
+      <option name="_new_additionalArguments" value="&quot;&quot;" />
+      <option name="_new_target" value="&quot;src.main.api.tests.create_transfer_test.TestCreateTransfer.test_transfer_deposit&quot;" />
+      <option name="_new_targetType" value="&quot;PYTHON&quot;" />
+      <method v="2" />
+    </configuration>
+    <configuration name="Python tests for src.main.api.tests.create_user_test.TestCreateUser.test_create_user_valid" type="tests" factoryName="Autodetect" temporary="true" nameIsGenerated="true">
+      <module name="NewSwaggerTest" />
+      <option name="ENV_FILES" value="" />
+      <option name="INTERPRETER_OPTIONS" value="" />
+      <option name="PARENT_ENVS" value="true" />
+      <option name="SDK_HOME" value="" />
+      <option name="WORKING_DIRECTORY" value="$PROJECT_DIR$" />
+      <option name="IS_MODULE_SDK" value="true" />
+      <option name="ADD_CONTENT_ROOTS" value="true" />
+      <option name="ADD_SOURCE_ROOTS" value="true" />
+      <option name="_new_additionalArguments" value="&quot;&quot;" />
+      <option name="_new_target" value="&quot;src.main.api.tests.create_user_test.TestCreateUser.test_create_user_valid&quot;" />
+      <option name="_new_targetType" value="&quot;PYTHON&quot;" />
+      <method v="2" />
+    </configuration>
+    <configuration name="Python tests for src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit_senior" type="tests" factoryName="Autodetect" temporary="true" nameIsGenerated="true">
+      <module name="NewSwaggerTest" />
+      <option name="ENV_FILES" value="" />
+      <option name="INTERPRETER_OPTIONS" value="" />
+      <option name="PARENT_ENVS" value="true" />
+      <option name="SDK_HOME" value="" />
+      <option name="WORKING_DIRECTORY" value="$PROJECT_DIR$" />
+      <option name="IS_MODULE_SDK" value="true" />
+      <option name="ADD_CONTENT_ROOTS" value="true" />
+      <option name="ADD_SOURCE_ROOTS" value="true" />
+      <option name="_new_additionalArguments" value="&quot;&quot;" />
+      <option name="_new_target" value="&quot;src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit_senior&quot;" />
+      <option name="_new_targetType" value="&quot;PYTHON&quot;" />
+      <method v="2" />
+    </configuration>
+    <recent_temporary>
+      <list>
+        <item itemvalue="Python tests.Python tests for src.main.api.tests.repay_credit_test.TestRepayCredit.test_repay_credit_senior" />
+        <item itemvalue="Python tests.Python tests for src.main.api.tests.create_credit_request_test.TestCreateRequestCredit.test_request_credit_senior" />
+        <item itemvalue="Python tests.Python tests for src.main.api.tests.create_transfer_test.TestCreateTransfer.test_transfer_deposit" />
+        <item itemvalue="Python tests.Python tests for src.main.api.tests.create_deposit_test.TestCreateDepositTest.test_deposit" />
+        <item itemvalue="Python tests.Python tests for src.main.api.tests.create_user_test.TestCreateUser.test_create_user_valid" />
+      </list>
+    </recent_temporary>
+  </component>
+  <component name="SharedIndexes">
+    <attachedChunks>
+      <set>
+        <option value="bundled-python-sdk-f69e5c277ee4-cbfc7b97e6b2-com.jetbrains.pycharm.community.sharedIndexes.bundled-PC-252.27397.106" />
+      </set>
+    </attachedChunks>
+  </component>
+  <component name="TaskManager">
+    <task active="true" id="Default" summary="Default task">
+      <changelist id="a63f5bb8-0c26-4d2b-af29-c8932684b6db" name="Changes" comment="" />
+      <created>1765434921411</created>
+      <option name="number" value="Default" />
+      <option name="presentableId" value="Default" />
+      <updated>1765434921411</updated>
+    </task>
+    <servers />
+  </component>
+</project>
\ No newline at end of file
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/vcs.xml b/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1765434923175)
+++ b/.idea/vcs.xml	(date 1765434923175)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: src/main/api/tests/create_transfer_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/create_transfer_test.py b/src/main/api/tests/create_transfer_test.py
new file mode 100644
--- /dev/null	(date 1766062127396)
+++ b/src/main/api/tests/create_transfer_test.py	(date 1766062127396)
@@ -0,0 +1,148 @@
+import pytest
+
+from src.main.api.models.deposit_request import DepositRequest
+from src.main.api.models.transfer_request import TransferRequest
+
+@pytest.mark.api
+class TestCreateTransfer:
+    # username = 'Stepans041122'
+    # def test_transfer_deposit(self):
+    #     login_admin_request = LoginUserRequest(username='admin', password='123456')
+    #     login_admin_response = requests.post(
+    #         url='http://localhost:4111/api/auth/token/login',
+    #         json=login_admin_request.model_dump(),
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'accept': 'application/json'
+    #         }
+    #     )
+    #     assert login_admin_response.status_code == 200
+    #     token = login_admin_response.json().get('token')
+    #
+    #     create_user_request = CreateUserRequest(username=TestCreateTransfer.username, password='Pas!sw0rd', role='ROLE_USER')
+    #     create_user_response = requests.post(
+    #         url = 'http://localhost:4111/api/admin/create',
+    #         json = create_user_request.model_dump(),
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'Authorization': f'Bearer {token}'
+    #         }
+    #     )
+    #     assert create_user_response.status_code == 200
+    #
+    #     login_user_request = LoginUserRequest(username=TestCreateTransfer.username, password='Pas!sw0rd')
+    #     login_user_response = requests.post(
+    #         url='http://localhost:4111/api/auth/token/login',
+    #         json=login_user_request.model_dump(),
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'accept': 'application/json',
+    #         }
+    #     )
+    #     assert login_user_response.status_code == 200
+    #     user_token = login_user_response.json().get('token')
+    #
+    #     create_from_deposit_response = requests.post(
+    #         url='http://localhost:4111/api/account/create',
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}'
+    #         }
+    #     )
+    #     assert create_from_deposit_response.status_code == 201
+    #     from_account_id = create_from_deposit_response.json().get('id')
+    #
+    #     create_to_deposit_response = requests.post(
+    #         url='http://localhost:4111/api/account/create',
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}'
+    #         }
+    #     )
+    #     assert create_to_deposit_response.status_code == 201
+    #     to_account_id = create_to_deposit_response.json().get('id')
+    #
+    #     create_deposit_request = DepositRequest(accountId=from_account_id, amount=5000.5)
+    #     deposit_response = requests.post(
+    #         url='http://localhost:4111/api/account/deposit',
+    #         json=create_deposit_request.model_dump(),
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}',
+    #             'Content-Type': 'application/json',
+    #         }
+    #     )
+    #     assert deposit_response.status_code == 200, f'Deposit failed: {deposit_response.text}'
+    #     balance = deposit_response.json().get('balance')
+    #     assert balance == 5000.5
+    #
+    #     create_transfer_request = TransferRequest(fromAccountId=from_account_id, toAccountId=to_account_id, amount=1250.38)
+    #     response = requests.post(
+    #         url='http://localhost:4111/api/account/transfer',
+    #         json=create_transfer_request.model_dump(),
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}',
+    #             'Content-Type': 'application/json',
+    #         }
+    #     )
+    #     assert response.status_code == 200, f'Transfer failed: {response.text}'
+    #     create_transfer_response = TransferResponse(**response.json())
+    #     assert create_transfer_response.fromAccountIdBalance == balance - create_transfer_request.amount
+
+
+    # def test_transfer_deposit_middle(self):
+    #     """Логинимся под администратора и создаем пользователя"""
+    #     create_user_request = CreateUserRequest(username=TestCreateTransfer.username,
+    #                                             password='Pas!sw0rd', role='ROLE_USER')
+    #     create_user_response = CreateUserRequester(
+    #         request_spec=RequestSpecs.auth_headers(username='admin', password='123456'),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(create_user_request)
+    #
+    #     """Логинимся пользователем и создаем первый счет который будет пополняться"""
+    #     create_account_from_response = CreateAccountRequest(
+    #         request_spec=RequestSpecs.auth_headers(username=create_user_request.username,
+    #                                                password=create_user_request.password),
+    #         response_spec=ResponseSpecs.request_create()
+    #     ).post()
+    #     account_id_from = create_account_from_response.id
+    #
+    #     """Создаем втророй счет на который будем переводить деньги с первого"""
+    #     create_account_to_response = CreateAccountRequest(
+    #         request_spec=RequestSpecs.auth_headers(username=create_user_request.username,
+    #                                                password=create_user_request.password),
+    #         response_spec=ResponseSpecs.request_create()
+    #     ).post()
+    #     account_id_to = create_account_to_response.id
+    #
+    #     """Пополняем первый счет"""
+    #     deposit_request = DepositRequest(accountId=account_id_from, amount = 3000)
+    #     create_deposit_response = CreateDepositRequester(
+    #         request_spec=RequestSpecs.auth_headers(username=create_user_request.username,
+    #                                                password=create_user_request.password),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(deposit_request)
+    #     balance = create_deposit_response.balance
+    #
+    #     """Переводим деньги с первого счета на второй, проверяем балан второго счета"""
+    #     create_transfer_request = TransferRequest(fromAccountId=account_id_from, toAccountId=account_id_to, amount=1250.38)
+    #     create_transfer_response = CreateTransferRequester(
+    #         request_spec=RequestSpecs.auth_headers(username = create_user_request.username, password = create_user_request.password),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(create_transfer_request)
+    #
+    #     assert create_transfer_response.fromAccountIdBalance == balance - create_transfer_request.amount
+    #     print(create_transfer_response.fromAccountIdBalance)
+
+    def test_transfer_deposit(self, api_manager, create_user_request):
+        from_account_id = api_manager.user_steps.create_account(create_user_request).id
+        deposit_request = DepositRequest(accountId=from_account_id, amount=5000)
+        deposit = api_manager.user_steps.create_deposit(create_user_request, deposit_request)
+
+        to_account_id = api_manager.user_steps.create_account(create_user_request).id
+        create_transfer = TransferRequest(fromAccountId=from_account_id, toAccountId=to_account_id, amount=3000)
+
+        transfer_response = api_manager.user_steps.create_transfer(create_user_request, create_transfer)
+
+        assert transfer_response.fromAccountIdBalance == 2000
\ No newline at end of file
Index: src/main/api/tests/create_credit_request_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/create_credit_request_test.py b/src/main/api/tests/create_credit_request_test.py
new file mode 100644
--- /dev/null	(date 1766062127414)
+++ b/src/main/api/tests/create_credit_request_test.py	(date 1766062127414)
@@ -0,0 +1,112 @@
+import pytest
+from src.main.api.fixtures.api_fixture import api_manager
+from src.main.api.models.credit_request import CreditRequest
+
+@pytest.mark.api
+class TestCreateRequestCredit:
+    # username = 'Stepans0211112'
+    # def test_request_credit(self):
+    #     login_admin_response = requests.post(
+    #         url='http://localhost:4111/api/auth/token/login',
+    #         json={
+    #             'username': 'admin',
+    #             'password': '123456'
+    #         },
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'accept': 'application/json'
+    #         }
+    #     )
+    #     assert login_admin_response.status_code == 200
+    #     token = login_admin_response.json().get('token')
+    #
+    #     create_user_response = requests.post(
+    #         url='http://localhost:4111/api/admin/create',
+    #         json={
+    #             'username': TestCreateRequestCredit.username,
+    #             'password': 'Pas!sw0rd',
+    #             'role': 'ROLE_CREDIT_SECRET',
+    #         },
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'Authorization': f'Bearer {token}'
+    #         }
+    #     )
+    #     assert create_user_response.status_code == 200
+    #
+    #     login_user_response = requests.post(
+    #         url='http://localhost:4111/api/auth/token/login',
+    #         json={
+    #             'username': TestCreateRequestCredit.username,
+    #             'password': 'Pas!sw0rd'
+    #         },
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'accept': 'application/json',
+    #         }
+    #     )
+    #     assert login_user_response.status_code == 200
+    #     user_token = login_user_response.json().get('token')
+    #
+    #     create_from_deposit_response = requests.post(
+    #         url='http://localhost:4111/api/account/create',
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}'
+    #         }
+    #     )
+    #     assert create_from_deposit_response.status_code == 201
+    #     from_account_id = create_from_deposit_response.json().get('id')
+    #
+    #     create_credit_response = requests.post(
+    #         url = 'http://localhost:4111/api/credit/request',
+    #         json = {
+    #             "accountId": from_account_id,
+    #             "amount": 5000,
+    #             "termMonths": 12
+    #         },
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}',
+    #             'Content-Type': 'application/json'
+    #         }
+    #     )
+    #
+    #     assert create_credit_response.status_code == 201, f'Create credit failed: {create_credit_response.text}'
+    #     assert create_credit_response.json().get('balance') == 5000
+    #     assert create_credit_response.json().get('termMonths') == 12
+
+    # def test_request_credit_middle(self):
+    #     """Логинимся под администратора и создаем пользователя"""
+    #     create_user_request = CreateUserRequest(username=TestCreateRequestCredit.username,
+    #                                             password='Pas!sw0rd', role='ROLE_CREDIT_SECRET')
+    #     create_user_response = CreateUserRequester(
+    #         request_spec=RequestSpecs.auth_headers(username='admin', password='123456'),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(create_user_request)
+    #     print(create_user_response.model_dump())
+    #
+    #     """Логинимся пользователем и создаем счет"""
+    #     create_account_response = CreateAccountRequest(
+    #         request_spec=RequestSpecs.auth_headers(username=create_user_request.username,
+    #                                                password=create_user_request.password),
+    #         response_spec=ResponseSpecs.request_create()
+    #     ).post()
+    #     account_id = create_account_response.id
+    #     print(create_account_response.model_dump())
+    #
+    #     """Подаем заявку на взятие кредита"""
+    #     create_credit_request = CreditRequest(accountId=account_id, amount=10000, termMonths=6)
+    #     create_credit_response = CreateCreditRequester(
+    #         request_spec=RequestSpecs.auth_headers(username=create_user_request.username, password=create_user_request.password),
+    #         response_spec=ResponseSpecs.request_create()
+    #     ).post(create_credit_request)
+    #
+    #     print(f'response error: {create_credit_request.model_dump()}')
+    #     assert create_credit_response.balance == 10000
+
+    def test_request_credit_senior(self, api_manager, create_user_request_credit):
+        account_id = api_manager.user_steps.create_account(create_user_request_credit).id
+        credit_request = CreditRequest(accountId=account_id, termMonths=12, amount=5000)
+        response = api_manager.user_steps.create_credit_request(create_user_request_credit, credit_request)
+        assert response.balance == 5000
\ No newline at end of file
Index: src/main/api/tests/repay_credit_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/repay_credit_test.py b/src/main/api/tests/repay_credit_test.py
new file mode 100644
--- /dev/null	(date 1766062127405)
+++ b/src/main/api/tests/repay_credit_test.py	(date 1766062127405)
@@ -0,0 +1,135 @@
+import pytest
+from src.main.api.models.credit_request import CreditRequest
+from src.main.api.models.repay_credit_request import RepayCreditRequest
+
+
+@pytest.mark.api
+class TestRepayCredit:
+    # username = 'Stepans0611222'
+    # def test_repay_credit(self):
+    #     login_admin_response = requests.post(
+    #         url='http://localhost:4111/api/auth/token/login',
+    #         json={
+    #             'username': 'admin',
+    #             'password': '123456'
+    #         },
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'accept': 'application/json'
+    #         }
+    #     )
+    #     assert login_admin_response.status_code == 200
+    #     token = login_admin_response.json().get('token')
+    #
+    #     create_user_response = requests.post(
+    #         url='http://localhost:4111/api/admin/create',
+    #         json={
+    #             'username': TestRepayCredit.username,
+    #             'password': 'Pas!sw0rd',
+    #             'role': 'ROLE_CREDIT_SECRET',
+    #         },
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'Authorization': f'Bearer {token}'
+    #         }
+    #     )
+    #     assert create_user_response.status_code == 200
+    #
+    #     login_user_response = requests.post(
+    #         url='http://localhost:4111/api/auth/token/login',
+    #         json={
+    #             'username': TestRepayCredit.username,
+    #             'password': 'Pas!sw0rd'
+    #         },
+    #         headers={
+    #             'Content-Type': 'application/json',
+    #             'accept': 'application/json',
+    #         }
+    #     )
+    #     assert login_user_response.status_code == 200
+    #     user_token = login_user_response.json().get('token')
+    #
+    #     create_from_deposit_response = requests.post(
+    #         url='http://localhost:4111/api/account/create',
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}'
+    #         }
+    #     )
+    #     assert create_from_deposit_response.status_code == 201
+    #     from_account_id = create_from_deposit_response.json().get('id')
+    #
+    #     create_credit_response = requests.post(
+    #         url = 'http://localhost:4111/api/credit/request',
+    #         json = {
+    #             "accountId": from_account_id,
+    #             "amount": 5000,
+    #             "termMonths": 12
+    #         },
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}',
+    #             'Content-Type': 'application/json'
+    #         }
+    #     )
+    #
+    #     assert create_credit_response.status_code == 201, f'Create credit failed: {create_credit_response.text}'
+    #     assert create_credit_response.json().get('balance') == 5000
+    #     assert create_credit_response.json().get('termMonths') == 12
+    #     credit_id = create_credit_response.json().get('creditId')
+    #
+    #     repay_credit = requests.post(
+    #         url='http://localhost:4111/api/credit/repay',
+    #         json={
+    #             "creditId": credit_id,
+    #             "accountId": from_account_id,
+    #             "amount": 5000
+    #         },
+    #         headers={
+    #             'accept': 'application/json',
+    #             'Authorization': f'Bearer {user_token}',
+    #             'Content-Type': 'application/json'
+    #         }
+    #     )
+    #     assert repay_credit.status_code == 200
+
+    # def test_repay_credit_level_2(self):
+    #     """Логинимся под администратора и создаем пользователя"""
+    #     create_user_requester = CreateUserRequest(username=TestRepayCredit.username, password='Pas!sw0rd', role='ROLE_CREDIT_SECRET')
+    #     create_user_response = CreateUserRequester(
+    #         request_spec=RequestSpecs.auth_headers(username='admin', password='123456'),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(create_user_requester)
+    #
+    #     '''Логинимся пользователем и создаем счет'''
+    #     create_account_requester = CreateAccountRequest(
+    #         request_spec=RequestSpecs.auth_headers(username=TestRepayCredit.username, password='Pas!sw0rd'),
+    #         response_spec=ResponseSpecs.request_create(),
+    #     ).post()
+    #     account_id = create_account_requester.id
+    #
+    #     """Подаем заявку на креди"""
+    #     create_credit_request = CreditRequest(accountId=account_id, amount=10000, termMonths=12)
+    #     credit_response = CreateCreditRequester(
+    #         request_spec=RequestSpecs.auth_headers(username=TestRepayCredit.username, password='Pas!sw0rd'),
+    #         response_spec=ResponseSpecs.request_create()
+    #     ).post(create_credit_request)
+    #     credit_id = credit_response.creditId
+    #
+    #     """Погашаем кредит единоразовым палтежем"""
+    #     create_repay_request = RepayCreditRequest(creditId=credit_id, accountId=account_id, amount=10000)
+    #     create_repay_response = CreateRepayRequester(
+    #         request_spec=RequestSpecs.auth_headers(username=TestRepayCredit.username, password='Pas!sw0rd'),
+    #         response_spec=ResponseSpecs.request_ok()
+    #     ).post(create_repay_request)
+    #
+    #     print(create_repay_response.model_dump())
+
+    def test_repay_credit_senior(self, api_manager, create_user_request_credit):
+        user_credit=api_manager.user_steps.create_account(create_user_request_credit)
+        credit_request=CreditRequest(accountId=user_credit.id, termMonths=12, amount=5000)
+        credit = api_manager.user_steps.create_credit_request(create_user_request_credit, credit_request)
+        repay_credit = RepayCreditRequest(creditId=credit.creditId, accountId=user_credit.id, amount=5000)
+        repay_response = api_manager.user_steps.repay_credit(create_user_request_credit, repay_credit)
+        print(repay_response)
+
Index: .idea/inspectionProfiles/profiles_settings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/inspectionProfiles/profiles_settings.xml b/.idea/inspectionProfiles/profiles_settings.xml
new file mode 100644
--- /dev/null	(date 1765434923148)
+++ b/.idea/inspectionProfiles/profiles_settings.xml	(date 1765434923148)
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
+    <option name="USE_PROJECT_PROFILE" value="false" />
+    <version value="1.0" />
+  </settings>
+</component>
\ No newline at end of file
Index: src/main/api/models/deposit_request.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/deposit_request.py b/src/main/api/models/deposit_request.py
new file mode 100644
--- /dev/null	(date 1765795352829)
+++ b/src/main/api/models/deposit_request.py	(date 1765795352829)
@@ -0,0 +1,5 @@
+from src.main.api.models.base_model import BaseModel
+
+class DepositRequest(BaseModel):
+    accountId: int
+    amount: float | int
\ No newline at end of file
Index: src/main/api/models/deposit_response.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/deposit_response.py b/src/main/api/models/deposit_response.py
new file mode 100644
--- /dev/null	(date 1765626154680)
+++ b/src/main/api/models/deposit_response.py	(date 1765626154680)
@@ -0,0 +1,6 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class DepositResponse(BaseModel):
+    id: int
+    balance: float | int
\ No newline at end of file
Index: src/main/api/models/transfer_request.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/transfer_request.py b/src/main/api/models/transfer_request.py
new file mode 100644
--- /dev/null	(date 1765626266768)
+++ b/src/main/api/models/transfer_request.py	(date 1765626266768)
@@ -0,0 +1,6 @@
+from src.main.api.models.base_model import BaseModel
+
+class TransferRequest(BaseModel):
+    fromAccountId: int
+    toAccountId: int
+    amount: float | int
\ No newline at end of file
Index: src/main/api/models/transfer_response.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/transfer_response.py b/src/main/api/models/transfer_response.py
new file mode 100644
--- /dev/null	(date 1765626458718)
+++ b/src/main/api/models/transfer_response.py	(date 1765626458718)
@@ -0,0 +1,8 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class TransferResponse(BaseModel):
+    fromAccountId: int
+    toAccountId: int
+    fromAccountIdBalance: float|int
+    
\ No newline at end of file
Index: src/main/api/models/credit_request.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/credit_request.py b/src/main/api/models/credit_request.py
new file mode 100644
--- /dev/null	(date 1765626623191)
+++ b/src/main/api/models/credit_request.py	(date 1765626623191)
@@ -0,0 +1,7 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class CreditRequest(BaseModel):
+    accountId: int
+    amount: float|int
+    termMonths: int
\ No newline at end of file
Index: src/main/api/models/credit_response.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/credit_response.py b/src/main/api/models/credit_response.py
new file mode 100644
--- /dev/null	(date 1765818881342)
+++ b/src/main/api/models/credit_response.py	(date 1765818881342)
@@ -0,0 +1,10 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class CreditResponse(BaseModel):
+    id: int
+    amount: float|int
+    termMonths: int
+    balance: float|int
+    creditId: int
+    
\ No newline at end of file
Index: src/main/api/models/repay_credit_request.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/repay_credit_request.py b/src/main/api/models/repay_credit_request.py
new file mode 100644
--- /dev/null	(date 1765626802525)
+++ b/src/main/api/models/repay_credit_request.py	(date 1765626802525)
@@ -0,0 +1,7 @@
+from src.main.api.models.base_model import BaseModel
+
+
+class RepayCreditRequest(BaseModel):
+    creditId: int
+    accountId: int
+    amount: float|int
\ No newline at end of file
Index: src/main/api/models/repay_credit_response.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/repay_credit_response.py b/src/main/api/models/repay_credit_response.py
new file mode 100644
--- /dev/null	(date 1765819205363)
+++ b/src/main/api/models/repay_credit_response.py	(date 1765819205363)
@@ -0,0 +1,7 @@
+from src.main.api.models.base_model import BaseModel
+
+
+
+class RepayCreditResponse(BaseModel):
+    creditId:int
+    amountDeposited: float|int
\ No newline at end of file
Index: src/main/api/requests/deposit_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/deposit_requester.py b/src/main/api/requests/deposit_requester.py
new file mode 100644
--- /dev/null	(date 1765804201152)
+++ b/src/main/api/requests/deposit_requester.py	(date 1765804201152)
@@ -0,0 +1,18 @@
+from src.main.api.models.deposit_response import DepositResponse
+from src.main.api.requests.requester import Requester
+from src.main.api.models.deposit_request import DepositRequest
+from requests import Response
+import requests
+
+
+class CreateDepositRequester(Requester):
+    def post(self, create_deposit_request:DepositRequest) -> DepositResponse | Response:
+        url = f'{self.base_url}/account/deposit'
+        response = requests.post(
+            url = url,
+            json=create_deposit_request.model_dump(),
+            headers= self.headers
+        )
+        self.response_spec(response)
+        return DepositResponse(**response.json())
+
Index: src/main/api/requests/transfer_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/transfer_requester.py b/src/main/api/requests/transfer_requester.py
new file mode 100644
--- /dev/null	(date 1765807346857)
+++ b/src/main/api/requests/transfer_requester.py	(date 1765807346857)
@@ -0,0 +1,16 @@
+import requests
+from requests import Response
+from src.main.api.models.transfer_response import TransferResponse
+from src.main.api.models.transfer_request import TransferRequest
+from src.main.api.requests.requester import Requester
+
+class CreateTransferRequester(Requester):
+    def post(self, create_transfer_request:TransferRequest) -> TransferResponse | Response:
+        url = f'{self.base_url}/account/transfer'
+        response = requests.post(
+            url = url,
+            json = create_transfer_request.model_dump(),
+            headers=self.headers
+        )
+        self.response_spec(response)
+        return TransferResponse(**response.json())
\ No newline at end of file
Index: src/main/api/requests/credit_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/credit_requester.py b/src/main/api/requests/credit_requester.py
new file mode 100644
--- /dev/null	(date 1765808389411)
+++ b/src/main/api/requests/credit_requester.py	(date 1765808389411)
@@ -0,0 +1,17 @@
+from src.main.api.models.credit_request import CreditRequest
+from src.main.api.models.credit_response import CreditResponse
+from src.main.api.requests.requester import Requester
+import requests
+from requests import Response
+
+
+class CreateCreditRequester(Requester):
+    def post(self, create_credit_requester:CreditRequest) -> CreditResponse | Response:
+        url = f'{self.base_url}/credit/request'
+        response = requests.post(
+            url = url,
+            json = create_credit_requester.model_dump(),
+            headers = self.headers
+        )
+        self.response_spec(response)
+        return CreditResponse(**response.json())
\ No newline at end of file
Index: src/main/api/requests/repay_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/requests/repay_requester.py b/src/main/api/requests/repay_requester.py
new file mode 100644
--- /dev/null	(date 1765818315889)
+++ b/src/main/api/requests/repay_requester.py	(date 1765818315889)
@@ -0,0 +1,17 @@
+from src.main.api.models.repay_credit_request import RepayCreditRequest
+from src.main.api.models.repay_credit_response import RepayCreditResponse
+from src.main.api.requests.requester import Requester
+import requests
+from requests import Response
+
+
+class CreateRepayRequester(Requester):
+    def post(self, create_repay_request:RepayCreditRequest) -> RepayCreditResponse | Response:
+        url = f'{self.base_url}/credit/repay'
+        response = requests.post(
+            url = url,
+            json = create_repay_request.model_dump(),
+            headers=self.headers,
+        )
+        self.response_spec(response)
+        return RepayCreditResponse(**response.json())
Index: src/main/api/foundation/endpoint.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/foundation/endpoint.py b/src/main/api/foundation/endpoint.py
new file mode 100644
--- /dev/null	(date 1766044923452)
+++ b/src/main/api/foundation/endpoint.py	(date 1766044923452)
@@ -0,0 +1,75 @@
+from src.main.api.models.base_model import BaseModel
+from typing import Optional, Type
+from dataclasses import dataclass
+from enum import Enum
+
+from src.main.api.models.create_account_response import CreateAccountResponse
+from src.main.api.models.create_user_reesponse import CreateUserResponse
+from src.main.api.models.create_user_request import CreateUserRequest
+from src.main.api.models.credit_request import CreditRequest
+from src.main.api.models.credit_response import CreditResponse
+from src.main.api.models.deposit_request import DepositRequest
+from src.main.api.models.deposit_response import DepositResponse
+from src.main.api.models.login_user_request import LoginUserRequest
+from src.main.api.models.login_user_response import LoginUserResponse
+from src.main.api.models.repay_credit_request import RepayCreditRequest
+from src.main.api.models.repay_credit_response import RepayCreditResponse
+from src.main.api.models.transfer_request import TransferRequest
+from src.main.api.models.transfer_response import TransferResponse
+from src.main.api.requests.create_account_requester import CreateAccountRequest
+
+
+@dataclass
+class EndpointConfiguration:
+    url: str
+    request_model: Optional[Type[BaseModel]]
+    response_model: Optional[Type[BaseModel]]
+
+class Endpoint(Enum):
+    ADMIN_CREATE_USER = EndpointConfiguration(
+        request_model=CreateUserRequest,
+        url='/admin/create',
+        response_model=CreateUserResponse
+    )
+
+    ADMIN_DELETE_USER = EndpointConfiguration(
+        request_model=None,
+        url="/admin/users",
+        response_model=None
+    )
+
+    LOGIN_USER = EndpointConfiguration(
+        request_model=LoginUserRequest,
+        url='/auth/token/login',
+        response_model=LoginUserResponse
+    )
+
+    CREATE_ACCOUNT = EndpointConfiguration(
+        request_model=None,
+        url='/account/create',
+        response_model=CreateAccountResponse
+    )
+
+    DEPOSIT_ACCOUNT = EndpointConfiguration(
+        request_model=DepositRequest,
+        url='/account/deposit',
+        response_model= DepositResponse
+    )
+
+    TRANSFER_ACCOUNT = EndpointConfiguration(
+        request_model=TransferRequest,
+        url='/account/transfer',
+        response_model=TransferResponse
+    )
+
+    CREDIT_REQUEST = EndpointConfiguration(
+        request_model=CreditRequest,
+        url='/credit/request',
+        response_model=CreditResponse
+    )
+
+    REPAY_CREDIT = EndpointConfiguration(
+        request_model=RepayCreditRequest,
+        url='/credit/repay',
+        response_model=RepayCreditResponse
+    )
\ No newline at end of file
Index: src/main/api/foundation/http_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/foundation/http_requester.py b/src/main/api/foundation/http_requester.py
new file mode 100644
--- /dev/null	(date 1765958264416)
+++ b/src/main/api/foundation/http_requester.py	(date 1765958264416)
@@ -0,0 +1,9 @@
+from src.main.api.foundation.endpoint import Endpoint
+from typing import Dict, Callable
+
+
+class HttpRequester:
+    def __init__(self, request_spec:Dict, endpoint:Endpoint, response_spec:Callable):
+        self.request_spec = request_spec
+        self.endpoint = endpoint
+        self.response_spec = response_spec
\ No newline at end of file
Index: src/main/api/foundation/crud_endpoint.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/foundation/crud_endpoint.py b/src/main/api/foundation/crud_endpoint.py
new file mode 100644
--- /dev/null	(date 1765884867158)
+++ b/src/main/api/foundation/crud_endpoint.py	(date 1765884867158)
@@ -0,0 +1,9 @@
+from typing import Protocol, Optional
+from requests import Response
+from src.main.api.models.base_model import BaseModel
+
+
+class CrudEndpoint(Protocol):
+    def post(self, model: Optional[BaseModel]) -> BaseModel | Response:...
+    def get(self, user_id: int) -> BaseModel | Response:...
+    def delete(self, user_id: int) -> BaseModel | Response:...
\ No newline at end of file
Index: src/main/api/foundation/requesters/crud_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/foundation/requesters/crud_requester.py b/src/main/api/foundation/requesters/crud_requester.py
new file mode 100644
--- /dev/null	(date 1765960491211)
+++ b/src/main/api/foundation/requesters/crud_requester.py	(date 1765960491211)
@@ -0,0 +1,26 @@
+from typing import Optional
+from requests import Response
+from src.main.api.configs.config import Config
+from src.main.api.foundation.http_requester import HttpRequester
+from src.main.api.models.base_model import BaseModel
+import requests
+
+
+class CrudRequester(HttpRequester):
+    def post(self, model: Optional[BaseModel]) -> Response:
+        body = model.model_dump() if model is not None else ""
+        response = requests.post(
+            url=f'{Config.fetch('backendUrl')}{self.endpoint.value.url}',
+            headers=self.request_spec,
+            json=body,
+        )
+        self.response_spec(response)
+        return response
+
+    def delete(self, user_id:int) -> Response:
+        response = requests.delete(
+            url= f'{Config.fetch('backendUrl')}{self.endpoint.value.url}/{user_id}',
+            headers=self.request_spec
+        )
+        self.response_spec(response)
+        return response
Index: src/main/api/foundation/requesters/validate_crud_requester.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/foundation/requesters/validate_crud_requester.py b/src/main/api/foundation/requesters/validate_crud_requester.py
new file mode 100644
--- /dev/null	(date 1766041332311)
+++ b/src/main/api/foundation/requesters/validate_crud_requester.py	(date 1766041332311)
@@ -0,0 +1,25 @@
+from typing import Optional
+
+from src.main.api.foundation.http_requester import HttpRequester
+from src.main.api.foundation.requesters.crud_requester import CrudRequester
+from src.main.api.models.base_model import BaseModel
+from typing import Optional
+
+class ValidateCrudeRequester(HttpRequester):
+    def __init__(self, request_spec, endpoint, response_spec):
+        super().__init__(request_spec, endpoint, response_spec)
+        self.crud_requester = CrudRequester(
+            request_spec = request_spec,
+            endpoint = endpoint,
+            response_spec = response_spec,
+        )
+
+    def post(self, model:Optional[BaseModel] = None) -> BaseModel:
+        response = self.crud_requester.post(model)
+        self.response_spec(response)
+        return self.endpoint.value.response_model.model_validate(response.json())
+
+    def delete(self, user_id: int):
+        response = self.crud_requester.delete(user_id)
+        self.response_spec(response)
+        return self.endpoint.value.response_model.model_validate(response.json())
\ No newline at end of file
Index: src/main/api/steps/admin_steps.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/steps/admin_steps.py b/src/main/api/steps/admin_steps.py
new file mode 100644
--- /dev/null	(date 1765975502666)
+++ b/src/main/api/steps/admin_steps.py	(date 1765975502666)
@@ -0,0 +1,43 @@
+
+from src.main.api.foundation.endpoint import Endpoint
+from src.main.api.foundation.requesters.validate_crud_requester import ValidateCrudeRequester
+from src.main.api.models.login_user_request import LoginUserRequest
+from src.main.api.requests.create_user_requester import CreateUserRequest
+from src.main.api.steps.base_steps import BaseSteps
+from src.main.api.specs.response_specs import ResponseSpecs
+from src.main.api.specs.request_specs import RequestSpecs
+from src.main.api.foundation.requesters.crud_requester import CrudRequester
+
+
+class AdminSteps(BaseSteps):
+    def create_user(self, create_user_request: CreateUserRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.auth_headers(username='admin', password='123456'),
+            Endpoint.ADMIN_CREATE_USER,
+            ResponseSpecs.request_ok()
+        ).post(create_user_request)
+
+        self.create_obj.append(response)
+        return response
+
+    def delete_user(self, user_id: int):
+        CrudRequester(
+            RequestSpecs.auth_headers(username='admin', password='123456'),
+            Endpoint.ADMIN_DELETE_USER,
+            ResponseSpecs.request_ok()
+        ).delete(user_id)
+
+    def create_invalid_user(self, create_user_request:CreateUserRequest):
+        CrudRequester(
+            RequestSpecs.auth_headers(username='admin', password='123456'),
+            Endpoint.ADMIN_CREATE_USER,
+            ResponseSpecs.request_bad()
+        ).post(create_user_request)
+
+    def login_user(self, login_user_request: LoginUserRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.unath_headers(),
+            Endpoint.LOGIN_USER,
+            ResponseSpecs.request_ok()
+        ).post(login_user_request)
+        return response
Index: src/main/api/steps/base_steps.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/steps/base_steps.py b/src/main/api/steps/base_steps.py
new file mode 100644
--- /dev/null	(date 1765961534464)
+++ b/src/main/api/steps/base_steps.py	(date 1765961534464)
@@ -0,0 +1,7 @@
+from typing import List, Any
+
+
+
+class BaseSteps:
+    def __init__(self, created_obj: List[Any]):
+        self.create_obj = created_obj
\ No newline at end of file
Index: src/main/api/steps/user_steps.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/steps/user_steps.py b/src/main/api/steps/user_steps.py
new file mode 100644
--- /dev/null	(date 1766056797553)
+++ b/src/main/api/steps/user_steps.py	(date 1766056797553)
@@ -0,0 +1,56 @@
+
+from src.main.api.fixtures.user_fixture import create_user_request
+from src.main.api.foundation.endpoint import Endpoint
+from src.main.api.foundation.requesters.validate_crud_requester import ValidateCrudeRequester
+from src.main.api.models.create_user_request import CreateUserRequest
+from src.main.api.models.credit_request import CreditRequest
+from src.main.api.models.deposit_request import DepositRequest
+from src.main.api.models.repay_credit_request import RepayCreditRequest
+from src.main.api.models.transfer_request import TransferRequest
+from src.main.api.specs.request_specs import RequestSpecs
+from src.main.api.specs.response_specs import ResponseSpecs
+from src.main.api.steps.base_steps import BaseSteps
+
+
+class UserSteps(BaseSteps):
+    def create_account(self, create_user_request: CreateUserRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.auth_headers(username=create_user_request.username, password=create_user_request.password),
+            Endpoint.CREATE_ACCOUNT,
+            ResponseSpecs.request_create()
+        ).post()
+        return response
+
+    def create_deposit(self, create_user_request: CreateUserRequest, deposit_request:DepositRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.auth_headers(username=create_user_request.username, password=create_user_request.password),
+            Endpoint.DEPOSIT_ACCOUNT,
+            ResponseSpecs.request_ok()
+        ).post(deposit_request)
+        return response
+
+    def create_transfer(self, create_user_request: CreateUserRequest, transfer_request:TransferRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.auth_headers(username=create_user_request.username, password=create_user_request.password),
+            Endpoint.TRANSFER_ACCOUNT,
+            ResponseSpecs.request_ok()
+        ).post(transfer_request)
+        return response
+
+    def create_credit_request(self, create_user_request_credit:CreateUserRequest, credit_request:CreditRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.auth_headers(username=create_user_request_credit.username, password=create_user_request_credit.password),
+            Endpoint.CREDIT_REQUEST,
+            ResponseSpecs.request_create()
+        ).post(credit_request)
+        return response
+
+    def repay_credit(self, create_user_request_credit: CreateUserRequest, repay_credit:RepayCreditRequest):
+        response = ValidateCrudeRequester(
+            RequestSpecs.auth_headers(username=create_user_request_credit.username, password=create_user_request_credit.password),
+            Endpoint.REPAY_CREDIT,
+            ResponseSpecs.request_ok()
+        ).post(repay_credit)
+        return response
+
+
Index: src/main/api/classes/api_manager.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/classes/api_manager.py b/src/main/api/classes/api_manager.py
new file mode 100644
--- /dev/null	(date 1765890088685)
+++ b/src/main/api/classes/api_manager.py	(date 1765890088685)
@@ -0,0 +1,9 @@
+from typing import List, Any
+from src.main.api.steps.admin_steps import AdminSteps
+from src.main.api.steps.user_steps import UserSteps
+
+
+class ApiManager:
+    def __init__(self, created_obj: List[Any]):
+        self.admin_steps = AdminSteps(created_obj)
+        self.user_steps = UserSteps(created_obj)
\ No newline at end of file
Index: src/main/api/fixtures/api_fixture.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/fixtures/api_fixture.py b/src/main/api/fixtures/api_fixture.py
new file mode 100644
--- /dev/null	(date 1766046631201)
+++ b/src/main/api/fixtures/api_fixture.py	(date 1766046631201)
@@ -0,0 +1,6 @@
+import pytest
+from src.main.api.classes.api_manager import ApiManager
+
+@pytest.fixture
+def api_manager(create_obj):
+    return ApiManager(create_obj)
\ No newline at end of file
Index: src/main/api/fixtures/object_fixture.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/fixtures/object_fixture.py b/src/main/api/fixtures/object_fixture.py
new file mode 100644
--- /dev/null	(date 1765962592030)
+++ b/src/main/api/fixtures/object_fixture.py	(date 1765962592030)
@@ -0,0 +1,20 @@
+import logging
+import pytest
+from typing import List, Any
+from src.main.api.classes.api_manager import ApiManager
+from src.main.api.models.create_user_reesponse import CreateUserResponse
+
+
+@pytest.fixture
+def create_obj():
+    objects: List[Any] = []
+    yield objects
+    clean_user(objects)
+
+def clean_user(objects: List[Any]):
+    api_manager = ApiManager(objects)
+    for i in objects:
+        if isinstance(i, CreateUserResponse):
+            api_manager.admin_steps.delete_user(i.id)
+        else:
+            logging.warning(f'Error in delete user_id: {i.id}')
\ No newline at end of file
Index: src/main/api/fixtures/user_fixture.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/fixtures/user_fixture.py b/src/main/api/fixtures/user_fixture.py
new file mode 100644
--- /dev/null	(date 1766061258236)
+++ b/src/main/api/fixtures/user_fixture.py	(date 1766061258236)
@@ -0,0 +1,17 @@
+import pytest
+
+from src.main.api.generators.model_generator import RandomModelGenerator
+from src.main.api.models.create_user_request import CreateUserRequest
+
+@pytest.fixture
+def create_user_request(api_manager):
+    user_request = RandomModelGenerator.generate(CreateUserRequest)
+    api_manager.admin_steps.create_user(user_request)
+    return user_request
+
+@pytest.fixture(scope='function')
+def create_user_request_credit(api_manager):
+    user_request = RandomModelGenerator.generate(CreateUserRequest)
+    user_request.role = "ROLE_CREDIT_SECRET"
+    api_manager.admin_steps.create_user(user_request)
+    return user_request
Index: src/main/api/generators/creation_rule.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/generators/creation_rule.py b/src/main/api/generators/creation_rule.py
new file mode 100644
--- /dev/null	(date 1766042308561)
+++ b/src/main/api/generators/creation_rule.py	(date 1766042308561)
@@ -0,0 +1,6 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class CreationRule:
+    regex: str
\ No newline at end of file
Index: src/main/api/generators/model_generator.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/generators/model_generator.py b/src/main/api/generators/model_generator.py
new file mode 100644
--- /dev/null	(date 1766043582315)
+++ b/src/main/api/generators/model_generator.py	(date 1766043582315)
@@ -0,0 +1,60 @@
+from typing import Any, get_type_hints, get_origin, Annotated, get_args
+import rstr
+import uuid
+import random
+
+
+
+from src.main.api.generators.creation_rule import CreationRule
+
+
+class RandomModelGenerator:
+    @staticmethod
+    def generate(cls: type) -> Any:
+        type_hints = get_type_hints(cls, include_extras=True)
+        init_data = {}
+
+        for field_name, annotated_type in type_hints.items():
+            rule = None
+            actual_type = annotated_type()
+
+            if get_origin(annotated_type) is Annotated:
+                actual_type, *annotations = get_args(annotated_type)
+                for ann in annotations:
+                    if isinstance(ann, CreationRule):
+                        rule = ann
+
+            if rule:
+                value = RandomModelGenerator._generate_from_regex(rule.regex, actual_type)
+            else:
+                value = RandomModelGenerator._generate_value(actual_type)
+
+            init_data[field_name] = value
+
+        return cls(**init_data)
+
+
+    @staticmethod
+    def _generate_from_regex(regex: str, field_type):
+        generated = rstr.xeger(regex)
+        if field_type is int:
+            return int(generated)
+        if field_type is float:
+            return float(generated)
+        return generated
+
+    @staticmethod
+    def _generate_value(field_type) -> Any:
+        if field_type is str:
+            return str(uuid.uuid4())[:8]
+        elif field_type is int:
+            return random.randint(1, 9999)
+        elif field_type is float:
+            return round(random.uniform(0, 100.), 2)
+        elif field_type is bool:
+            return random.choice([True, False])
+        elif field_type is list:
+            return [str(uuid.uuid4())[:5]]
+        elif isinstance(field_type, type):
+            return RandomModelGenerator.generate(field_type)
+        return None
\ No newline at end of file
Index: src/main/api/tests/create_user_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/create_user_test.py b/src/main/api/tests/create_user_test.py
new file mode 100644
--- /dev/null	(date 1766062127386)
+++ b/src/main/api/tests/create_user_test.py	(date 1766062127386)
@@ -0,0 +1,34 @@
+import pytest
+
+from src.main.api.generators.model_generator import RandomModelGenerator
+from src.main.api.models.create_user_request import CreateUserRequest
+
+@pytest.mark.api
+class TestCreateUser:
+    @pytest.mark.parametrize(
+        'create_user_request',
+        [RandomModelGenerator.generate(CreateUserRequest)],
+    )
+    def test_create_user_valid(self, api_manager, create_user_request):
+        response = api_manager.admin_steps.create_user(create_user_request)
+        print(response)
+        assert create_user_request.username == response.username
+        assert create_user_request.role == response.role
+
+    @pytest.mark.parametrize(
+        'username, password',
+        [
+            ('абв','Pas!sw0rd'),
+            ('аб', 'Pas!sw0rd'),
+            ('abv!', 'Pas!sw0rd'),
+            ('Stepan002', 'Pas!sw0rд'),
+            ('Stepan003', 'Pas!sw0'),
+            ('Stepan004', 'pas!sw0rd'),
+            ('Stepan005', 'PAS!SW0RD'),
+            ('Stepan006', 'PASSSWORD'),
+            ('Stepan007', 'PAS!SWRRD')
+        ]
+    )
+    def test_create_user_invalid(self, username, password, api_manager):
+        create_user_request = CreateUserRequest(username=username, password=password, role='ROLE_USER')
+        api_manager.admin_steps.create_invalid_user(create_user_request)
Index: src/main/api/tests/user_login_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/user_login_test.py b/src/main/api/tests/user_login_test.py
new file mode 100644
--- /dev/null	(date 1766040401967)
+++ b/src/main/api/tests/user_login_test.py	(date 1766040401967)
@@ -0,0 +1,19 @@
+import pytest
+from src.main.api.models.login_user_request import LoginUserRequest
+
+
+@pytest.mark.api
+class TestUserLogin:
+    def test_login_admin(self, api_manager):
+        login_admin_request = LoginUserRequest(username='admin', password='123456')
+        response = api_manager.admin_steps.login_user(login_admin_request)
+
+        assert login_admin_request.username ==  response.user.username
+        assert response.user.role == 'ROLE_ADMIN'
+
+    def test_user_login(self, api_manager, create_user_request):
+
+        response = api_manager.admin_steps.login_user(create_user_request)
+
+        assert create_user_request.username == response.user.username
+        assert response.user.role == 'ROLE_USER'
\ No newline at end of file
Index: src/main/api/tests/create_account_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/tests/create_account_test.py b/src/main/api/tests/create_account_test.py
new file mode 100644
--- /dev/null	(date 1766062056041)
+++ b/src/main/api/tests/create_account_test.py	(date 1766062056041)
@@ -0,0 +1,10 @@
+import pytest
+
+
+
+@pytest.mark.api
+class TestCreateAccount:
+    def test_create_account(self, api_manager, create_user_request):
+        response = api_manager.user_steps.create_account(create_user_request)
+        print(response)
+        assert response.balance == 0
Index: pytest.ini
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pytest.ini b/pytest.ini
new file mode 100644
--- /dev/null	(date 1765542976385)
+++ b/pytest.ini	(date 1765542976385)
@@ -0,0 +1,6 @@
+[pytest]
+
+markers=
+    api: Api Test
+
+pythonpath = .
\ No newline at end of file
Index: src/main/api/specs/request_specs.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/specs/request_specs.py b/src/main/api/specs/request_specs.py
new file mode 100644
--- /dev/null	(date 1765974945569)
+++ b/src/main/api/specs/request_specs.py	(date 1765974945569)
@@ -0,0 +1,34 @@
+
+from src.main.api.models.login_user_request import LoginUserRequest
+from src.main.api.models.login_user_response import LoginUserResponse
+import requests
+
+
+class RequestSpecs:
+    @staticmethod
+    def base_header():
+        return {
+            'Content-Type': 'application/json',
+            'accept': 'application/json'
+        }
+
+    @staticmethod
+    def auth_headers(username:str, password:str):
+        request = LoginUserRequest(username=username, password=password)
+        response = requests.post(
+            url = 'http://localhost:4111/api/auth/token/login',
+            json = request.model_dump(),
+            headers = RequestSpecs.base_header(),
+        )
+
+        if response.status_code == 200:
+            response_data = LoginUserResponse(**response.json())
+            token = response_data.token
+            headers = RequestSpecs.base_header()
+            headers['Authorization'] = f'Bearer {token}'
+            return headers
+        raise Exception('Failed to login')
+
+    @staticmethod
+    def unath_headers():
+        return RequestSpecs.base_header()
Index: src/main/api/specs/response_specs.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/specs/response_specs.py b/src/main/api/specs/response_specs.py
new file mode 100644
--- /dev/null	(date 1765443702451)
+++ b/src/main/api/specs/response_specs.py	(date 1765443702451)
@@ -0,0 +1,23 @@
+from requests import Response
+from http  import HTTPStatus
+
+
+
+class ResponseSpecs:
+    @staticmethod
+    def request_ok():
+        def confirm(response:Response):
+            assert response.status_code == HTTPStatus.OK, response.text
+        return confirm
+
+    @staticmethod
+    def request_create():
+        def confirm(response:Response):
+            assert response.status_code == HTTPStatus.CREATED, response.text
+        return confirm
+
+    @staticmethod
+    def request_bad():
+        def confirm(response:Response):
+            assert response.status_code == HTTPStatus.BAD_REQUEST,  response.text
+        return confirm
\ No newline at end of file
Index: src/main/api/models/login_user_request.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/api/models/login_user_request.py b/src/main/api/models/login_user_request.py
new file mode 100644
--- /dev/null	(date 1765627126602)
+++ b/src/main/api/models/login_user_request.py	(date 1765627126602)
@@ -0,0 +1,5 @@
+from src.main.api.models.base_model import BaseModel
+
+class LoginUserRequest(BaseModel):
+    username: str
+    password: str
\ No newline at end of file
